src/components/common/Modal.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'proptypes';
+import classNames from 'classnames';
+import { closeDateRangeModal } from '../../redux/reducers/data';
+
+const Modal = ({
+  open,
+  content,
+  style,
+  closeModal,
+}) => {
+  const modalClassName = classNames('modal', {
+    'is-active': open,
+  });
+
+  useEffect(() => {
+    const handleEscapeClick = (e) => {
+      if (e.keyCode === 27) {
+        closeModal();
+      }
+    };
+
+    if (open) {
+      document.addEventListener('keydown', handleEscapeClick);
+    } else {
+      document.removeEventListener('keydown', handleEscapeClick);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleEscapeClick);
+    };
+  }, [open, closeModal]);
+
+  return (
+    <div
+      className={modalClassName}
+      // This z-index is greater than the Menu component's z-index but less than
+      // the Header and Footer. The modal will cover the Menu when open, but not
+      // the Header or Footer.
+      style={{ zIndex: '10000', ...style }}
+    >
+      <div className="modal-background" />
+      <div className="modal-content" style={{ overflow: 'visible' }}>
+        {content}
+      </div>
+    </div>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  open: state.data.dateRangeModalOpen,
```

Comment:
I'm hesitant to store things like this in redux. If it doesn't need to be shared across the application can the parent manage the modal state?

Reply:
Removed all modal-related redux stuff. `<Modal />` ` open` prop is now controlled directly by its parent.